src/Plugin/migrate/destination/DgiRevisionedEntity.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Drupal\dgi_migrate\Plugin\migrate\destination;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\migrate\MigrateException;
+use Drupal\migrate\Plugin\migrate\destination\EntityContentBase;
+use Drupal\migrate\Plugin\MigrateIdMapInterface;
+use Drupal\migrate\Plugin\MigrationInterface;
+use Drupal\migrate\Row;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a custom destination plugin for generating entity revisions.
+ *
+ * This class extends the EntityContentBase class to provide a custom
+ * destination plugin for Drupal migrations. It allows for the creation of
+ * revisions during the migration process, and maintains revision history.
+ *
+ * @see \Drupal\migrate\Plugin\migrate\destination\EntityContentBase
+ *
+ * @MigrateDestination(
+ *   id = "dgi_revisioned_entity"
+ * )
+ */
+class DgiRevisionedEntity extends EntityContentBase {
+
+  /**
+   * The entity type.
+   *
+   * @var string
+   */
+  public static string $entityType;
```

Comment:
Should probably no longer be static. Could lead to confusion if ever instantiating multiple of these with different types in a single PHP process, as the type would be whichever was instantiated last.

```suggestion
  public string $entityType;
```

Reply:
Though, on second though, this value isn't really used? Could we just drop this variable entirely?